### PR TITLE
MWPW-133262: Adds localization functionality to custom filters

### PR DESCRIPTION
**Pull request by cmiqueo**
* Adds functionality to localize custom filters using the same process as the collection title

Resolves: [MWPW-133262](https://jira.corp.adobe.com/browse/MWPW-133262)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/tools/caas?martech=off
- After: [https://mwpw-133262--milo--adobecom.hlx.page/tools/caas?martech=off](https://mwpw-133262--milo--adobecom.hlx.page/tools/caas?martech=off)

To verify use this settings:
Mapping file: https://main--milo--adobecom.hlx.page/drafts/caas/mappings
Group Name: customGroupA
Custom Filter: customFilterA1
Custom Filter: customFilterA2
Group Name: customGroupB
Custom Filter: customFilterB1
Custom Filter: customFilterB2

expected result:
image

### CODE REVIEW

File: libs/blocks/caas/utils.js [hunk 1]
```diff
@@ -208,25 +208,25 @@ const getFilterObj = ({ excludeTags, filterTag, icon, openedOnLoad }, tags, stat
...
 };
 
-const getCustomFilterObj = ({ group, filtersCustomItems, openedOnLoad }) => {
+const getCustomFilterObj = ({ group, filtersCustomItems, openedOnLoad }, strs = {}) => {
   if (!group) return null;
 
   const items = filtersCustomItems.map((item) => ({
     id: item.customFilterTag[0],
-    label: item.filtersCustomLabel,
+    label: item.filtersCustomLabel?.match(/^{.*}$/) ? strs[item.filtersCustomLabel.replace(/{|}/g, '')] : item.filtersCustomLabel || '',
```
Comment (chrischrischris) [inline libs/blocks/caas/utils.js:216]: nit: get line length < 100

Comment (chrischrischris) [reply libs/blocks/caas/utils.js:216]: Since `/^{.*}$/` is used multiple times you can move it out to a const: `const IN_BRACKETS_RE = /^{.*}$/;`

**Review by chrischrischris — APPROVED** [review]
Some nits but otherwise 👍

**Review by overmyheadandbody — APPROVED** [review]

**Review by mokimo — APPROVED** [review]

**Comment by honstar** [comment]
Nit: the diff codecoverage check is failing, would you mind adding more tests to cover the referenced lines? Thx

**Review by elaineskpt — APPROVED** [review]

File: libs/blocks/caas/utils.js (hunk 1 above)
Comment (cmiqueo) [reply libs/blocks/caas/utils.js]: Done!

Comment (cmiqueo) [reply libs/blocks/caas/utils.js]: @chrischrischris, Can we merge this?

**Comment by honstar** [comment]
> Nit: the diff codecoverage check is failing, would you mind adding more tests to cover the referenced lines? Thx

This is still failing, please add more tests.

**Review by Ruchika4 — APPROVED** [review]

**Comment by auniverseaway** [comment]
@cmiqueo who can validate this as fixed? Typically we like to have an assignee approve.

**Pull request merged**